src/com/griddynamics/yvoronovskyi/battleship/BattleshipField.java

```diff
@@ -0,0 +1,231 @@
+
+            isShipCoordinateValid = (firstX == secondX || firstY == secondY);
+            if (!isShipCoordinateValid) {
+                System.out.println(Constants.WRONG_SHIP_LOCATION_MESSAGE);
+
+            }
+            int sDesk = (firstX == secondX)
+                    ? Math.abs(firstY - secondY)
+                    : Math.abs(firstX - secondX);
+            isShipCoordinateValid = sDesk + 1 == desk;
+            if (!isShipCoordinateValid) {
+                System.out.println(Constants.WRONG_SHIP_LENGTH_MESSAGE);
+                continue;
+            }
+            isShipCoordinateValid = setShipOnMap(firstX, firstY, secondX, secondY, desk);
+            if (!isShipCoordinateValid) {
+                System.out.println(Constants.WRONG_TOO_CLOSE_LOCATION_MESSAGE);
+                continue;
+            }
+            System.out.println(this.printDesk(Constants.PRINT_ALL));
+        }
+    }
+
+    public boolean setShipOnMap(int firstX, int firstY, int secondX, int secondY, int desk) {
+        boolean isHorizontal = firstX == secondX;
+        int begin;
+        if (isHorizontal) {
+            begin = Math.min(firstY, secondY);
+            for (int index = begin; index < begin + desk; index++) {
+                if (!isOcupated(firstX, index)) {
+                    return false;
+                }
+            }
+            shipId++;
+            for (int index = begin; index < begin + desk; index++) {
+                makeOccupation(firstX, index, Constants.EMPTY_DELIMETER + shipId);
+            }
+        } else {
+            begin = Math.min(firstX, secondX);
+            for (int index = begin; index < begin + desk; index++) {
+                if (!isOcupated(index, firstY)) {
+                    return false;
+                }
+            }
+            shipId++;
+            for (int index = begin; index < begin + desk; index++) {
+                makeOccupation(index, firstY, Constants.EMPTY_DELIMETER + shipId);
+            }
+        }
+        return true;
+    }
+
+    private int getPrevious(int point) {
+        return (point == 0) ? point : point - 1;
+    }
+
+    private int getNext(int point) {
+        return (point == 9) ? point : point + 1;
+    }
+
+    private void makeOccupation(int x, int y, String shipId) {
+        fieldMap[x][y] = Constants.SHIP;
+        shipIdMap[x][y] = shipId;
+        for (int xIndex = getPrevious(x); xIndex <= getNext(x); xIndex++) {
+            for (int yIndex = getPrevious(y); yIndex <= getNext(y); yIndex++) {
+                occupationMap[xIndex][yIndex] = Constants.SHIP;
+            }
+        }
+    }
+
+    public Point getMapPoints(String mapPoints) {
+        char[] coordinate = mapPoints.toCharArray();
+        int x = Character.getNumericValue(coordinate[0]);
+        int y = (coordinate.length > 2)
+                ? Character.getNumericValue(coordinate[1] + coordinate[2])
```

Comment:
Can you explain, why do you add the number of first point to the letter of the second, e.g. A2 B4 -> 2 + B?

Reply:
Renamed method, also I am not sure that I can solved issue with out Point in this case.